rest_tools/client/client.py

```diff
@@ -177,6 +178,49 @@ def request_seq(self, method, path, args=None):
             r = self.session.request(method, url, **kwargs)
             r.raise_for_status()
             return self._decode(r.content)
+        except requests.exceptions.HTTPError as e:
+            if method == 'DELETE' and e.response.status_code == 404:
+                raise # skip the logging for an expected error
+            logging.info('bad request: %s %s %r', method, path, args, exc_info=True)
+            raise
+        finally:
+            self.session = s
+
+    def request_streaming(self, method, path, args=None):
+        """
+        Send request to REST Server, and stream results back.
+
+        Args:
+            method (str): the http method
+            path (str): the url path on the server
+            args (dict): any arguments to pass
+
+        Returns:
+            dict: json dict or raw string
+        """
+        s = self.session
+        try:
+            self.open(sync=True)
+            url, kwargs = self._prepare(method, path, args)
+            kwargs['timeout'] = 100000 # disable timeout for streaming
+            r = self.session.request(method, url, stream=True, **kwargs)
+            buf = ''
+            for line in r.iter_lines(chunk_size=4096, decode_unicode=True, delimiter='\n'):
+                buf += line+'\n'
+                if buf.strip():
```

Comment:
Is your intention here to skip white-space lines? This seems like a roundabout way of doing: `if not line.strip(): continue`

Reply:
I think this was just a test of "is there any json data in the buffer for me to process."  Thinking about it, you could also just skip every blank line, since you know a blank line will never close a json document.

Reply:
I suppose this way it does keep the concatenated new-line chars as non-leading/trailing, in case that's important for the buffer.